pages/Oven_Pricing.py

```diff
@@ -5,20 +5,134 @@
+                    color: white;
+                    text-align: center;
+                    padding: 20px;
+                    border-radius: 10px;
+                    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);">
+                    <strong style="font-size: 1.2em;">Current Oven Cost</strong><br>
+                    <span style="font-size: 1.4em;">£{cost_now:.4f}</span>
+                </div>
+                """,
+            unsafe_allow_html=True,
+        )
+
+
+def calculate_kettle_cost(current_price, next_price, run_time, power):
+    cost_now = ((run_time / 3600) * current_price * power) / 100
+    cost_next = ((run_time / 3600) * next_price * power) / 100
+    return cost_now, cost_next
+
+
+st.markdown(
+    """
+    <style>
+    /* Style for the number input label */
+    label[for="total-bake-time"] {
+        font-size: 1.5em; /* Increase label font size */
+        font-weight: bold; /* Make label bold */
+    }
+
+    /* Style for the number input box */
+    div[data-baseweb="input"] input {
+        font-size: 1.1em; /* Increase font size inside the input box */
+        padding: 10px; /* Add padding for a larger input field */
+        height: 60px; /* Set height for larger appearance */
+    }
+    </style>
+    """,
+    unsafe_allow_html=True,
+)
+
+
 def main():
-    st.title("Washing Machine Pricing on Octopus Agile")
+    st.title("Oven Pricing on Octopus Agile")
 
     if st.session_state.df is not None:
         toggle = st.toggle("Select time manually", False)
         current_time = get_current_time(toggle, st.session_state.df)
-        bake_time = st.number_input("Total Bake Time [min]:", value=20)
+
+        st.markdown(
+            f"""
+                <div style="
+                    color: {st.session_state.font};
+                    border-radius: 10px;
+                    padding-bottom: 10px;">
+                    <span style="font-size: 1.1em;">Total bake time [min]:</span><br>
+                </div>
+                """,
+            unsafe_allow_html=True,
+        )
+        bake_time = st.number_input(
+            "Total bake time [min]: ", value=20, label_visibility="collapsed"
+        )
         oven_power = 2.5  # kW
-        current_price, next_price, current_cost_row, next_cost_row = get_current_cost(
+        current_price, next_price, current_cost_row, _ = get_current_cost(
             st.session_state.df, current_time
         )
-        cost = oven_power * bake_time / 60 * current_price / 100
+        time_to_next_half_hour = 30 - current_time.minute % 30
+
+        if time_to_next_half_hour > bake_time:
+            cost = oven_power * (current_price * (bake_time) / 60) / 100  # £
+        else:
+            cost = (
+                oven_power * (current_price * (time_to_next_half_hour) / 60) / 100
+                + oven_power
+                * (next_price * (bake_time - time_to_next_half_hour) / 60)
+                / 100
+            )  # £
```

Comment:
I think this simplifies the equation (and hopefully doesn't change it).
```suggestion
            cost = oven_power * ((current_price * time_to_next_half_hour + next_price * (bake_time - time_to_next_half_hour))) / 600 
```